src/Validator/Constraints/MollieGatewayConfigValidator.php

```diff
@@ -26,54 +27,100 @@ final class MollieGatewayConfigValidator extends ConstraintValidator
 
     private const AMOUNT_LIMITS_FIELD = 'amountLimits';
 
-    private const FIELD_VALUE = 'value';
-
     public function validate(mixed $value, Constraint $constraint): void
     {
         Assert::isInstanceOf($constraint, MollieGatewayConfigValidatorType::class);
 
-        if ($value instanceof PersistentCollection) {
-            $this->validateAmounts($value, $constraint);
+        if (!$value instanceof PersistentCollection) {
+            return;
         }
-    }
 
-    private function validateAmounts(PersistentCollection $collection, MollieGatewayConfigValidatorType $constraint): void
-    {
-        $mollieGatewayConfigs = $collection->getSnapshot();
-        foreach ($mollieGatewayConfigs as $key => $mollieGatewayConfig) {
-            $amountLimits = $mollieGatewayConfig->getAmountLimits();
-            if (!$amountLimits) {
+        /** @var MollieGatewayConfigInterface[] $configs */
+        $configs = $value->getSnapshot();
+
+        foreach ($configs as $index => $config) {
+            $limits = $config->getAmountLimits();
+            if ($limits === null) {
                 continue;
             }
 
-            $minAmount = $amountLimits->getMinimumAmount();
-            $maxAmount = $amountLimits->getMaximumAmount();
+            $configMinimum = $limits->getMinimumAmount();
+            $configMaximum = $limits->getMaximumAmount();
```

Comment:
The same, why did you change the name of variables? But if we are changing it, I would be for:
```suggestion
            $minimumAmount = $limits->getMinimumAmount();
            $maximumAmount = $limits->getMaximumAmount();
```

Reply:
Working with minimumAmount and maximumAmount, which could come from either the limits configuration or the payment method configuration, kept confusing me over time. So, I decided to introduce naming that immediately clarifies which value we’re dealing with.

Reply:
Here are the methods I mentioned above:
https://github.com/Sylius/MolliePlugin/blob/3ce2bc8539a425744c5e56657a636d0e533b3504/src/Payments/Methods/ConfigTrait.php#L57
https://github.com/Sylius/MolliePlugin/blob/3ce2bc8539a425744c5e56657a636d0e533b3504/src/Entity/MollieMinMax.php#L37

Reply:
To avoid confusion between the value coming from the limit configuration and the value returned by the API, I’m prefixing variable names with `limit` and `api` to immediately signal each value’s source.